workbench-core/datasets-ui/src/components/CreateInternalDatasetForm.tsx

```diff
@@ -0,0 +1,282 @@
+        setFormErrors,
+        datasetProjectIdValidationRules,
+        formData.projectId
+      ) && isValid;
+
+    return isValid;
+  };
+
+  const submitForm = async (): Promise<void> => {
+    if (!validateAll()) {
+      return;
+    }
+
+    setIsSubmitLoading(true);
+    try {
+      await onFormSubmit(formData);
+    } catch {
+      setError('There was a problem creating a dataset.');
+    } finally {
+      setIsSubmitLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    // eslint-disable-next-line security/detect-object-injection
+    setDisableSubmit(Object.keys(formErrors).some((field) => !!formErrors[field]));
+  }, [formErrors]);
+
+  const getBasicDetailsSection = (): JSX.Element => {
+    return (
+      <ExpandableSection
+        defaultExpanded
+        variant="container"
+        header={
+          <Header
+            variant="h2"
+            description="This Dataset will be added to your Organizational datasets and can be shared according to permissions"
+          >
+            Internal Dataset Details
+          </Header>
+        }
+      >
+        <SpaceBetween direction="vertical" size="l">
+          <FormField
+            label="Dataset Name"
+            description="Create a unique name that you can reference easily in the Organization datasets dashboard"
+            constraintText={<>Alphanumeric characters and hyphens.</>}
+            errorText={formErrors.nameError}
+          >
+            <Input
+              data-testid="datasetName"
+              value={formData?.name || ''}
+              onChange={({ detail: { value } }) => {
+                setFormData({ ...formData, name: value });
+                validateField<string | undefined, CreateDatasetForm, CreateDatasetFormValidation>(
+                  'name',
+                  setFormErrors,
+                  datasetNameValidationRules,
+                  value
+                );
+              }}
+            />
+          </FormField>
+          <FormField
+            label={
+              <>
+                Dataset description - <i>optional</i>
+              </>
+            }
+            description="Enter a short description of your dataset. Max character limit 500."
+            errorText={formErrors.descriptionError}
+          >
+            <Textarea
+              data-testid="datasetDescription"
+              onChange={({ detail: { value } }) => {
+                setFormData({ ...formData, description: value });
+                validateField<string | undefined, CreateDatasetForm, CreateDatasetFormValidation>(
+                  'description',
+                  setFormErrors,
+                  datasetDescriptionValidationRules,
+                  value
+                );
+              }}
+              value={formData?.description || ''}
+              placeholder="Description"
+            />
+          </FormField>
+          <FormField
+            label="Owning Project"
+            // TODO: add href
```

Comment:
still needed?

Reply:
Yea, this is still WIP, we don't have behavior for this link